js/main.js

```diff
@@ -1,255 +1,4 @@
-            }
-            if (_.random(1, 100) <= 25) {
-                label += ' of the ' + _.sample(vocabulary.internet) + '.';
-            } else {
-                label += '.';
-            }
-    }
-    return label;
-};
-
-//
-// Good Morning Scum
-//
-// Produces a quote like: "Good Morning, Bastards of the Internet"
-//
-var goodMorningScum = function(vocabulary) {
-    var greeting = '';
-
-    // eg, Good Morning
-    greeting += _.sample(vocabulary.salutation) + ' ';
-
-    // eg, "bastards of the internet"
-    greeting += ellisLabel(vocabulary);
-
-    if (_.random(1, 100) <= 10) {
-        // eg, "I hate you"
-        greeting += ' ' + _.sample(vocabulary.extra);
-    }
-
-    return greeting;
-};
-
-//
-// Attention Scum
-//
-// Produces a quote like "ATTENTION SCUM: I will destroy you"
-//
-var attentionScum = function(vocabulary) {
-    var greeting = '';
-
-    if (_.random(1, 100) <= 50) {
-        // eg, "ATTENTION SCUM"
-        greeting += _.sample(vocabulary.attention) + ' ';
-
-    } else {
-        // eg, "ATTENTION $FILTH" with a random filth name
-        greeting += "attention ";
-        greeting += _.sample(vocabulary.attentive_filth) + ': ';
-    }
-
-    // uppercase the "ATTENTION SCUM" part
-    greeting = greeting.toUpperCase();
-
-    // eg, "I hate you"
-    greeting += _.sample(vocabulary.extra);
-
-    return greeting;
-};
-
-//
-// Build a Quote
-//
-// This is our final function. It chooses between two paths:
-// 1. 85% chance to call goodMorningScum(), and an
-//    addtional 10% chance to append an extra bit at the end.
-// 2. 15% chance to call attentionScum()
-//
-var buildQuote = function(vocabulary) {
-    var quote = '';
-
-    if (_.random(1, 100) <= 85) {
-        // eg, "Good Morning, Scum"
-        quote = goodMorningScum(vocabulary);
-
-    } else {
-        // eg, "ATTENTION SCUM: I hate you"
-        quote = attentionScum(vocabulary);
-    }
-
-    return quote;
-};
-
-// Prepare the Ajax request
-xhr.open('GET', 'data/vocabulary.json', true);
-
-// Handle the Ajax response
-xhr.onload = function() {
-
-    // We got a good response from the server
-    if (xhr.status >= 200 && xhr.status < 400) {
-
-        // Parse the JSON data
-        var vocabulary = JSON.parse(xhr.responseText);
-        document.getElementById('quote').innerHTML = buildQuote(vocabulary);
-
-    // We reached our target server, but it returned an error
-    } else {
-        var error = 'Whoops! Something went wrong. Please try again.';
-        document.getElementById('datelist-error').innerHTML = error;
-    }
-};
-
-// Handle an Ajax connection error
-xhr.onerror = function() {
-    var error = 'Whoops! We couldn\'t reach the server. Please try again.';
-    document.getElementById('datelist-error').innerHTML = error;
-};
-
-// Send the Ajax request
-xhr.send();
+var vocabulary = loadData('data/vocabulary.json');
+document.getElementById('quote').innerHTML = talkLikeWarrenEllis(vocabulary);
```

Comment:
Javascript is faux-async*, so L4 happens before L3. Usually this is fixed by changing main.js to something like 

```
loadData('data/vocabulary.json', function (vocabulary) {
  document.getElementById('quote').innerHTML = talkLikeWarrenEllis(vocabulary);
}
```

Inside `loadData` you'll change it to look something like:

```
var loadData = function(url, successCallback) {
  ...
  var vocabulary = JSON.parse(request.responseText);
  console.log(vocabulary);
  successCallback(vocabulary);
  ...
}
```

Buuuuut, now we get into Promises/Futures, which are probably a more straightforward approach, but I'm not sure they are the answer for VanillaJS. Unless you're using ES6 of course! http://www.datchley.name/es6-promises/

* I called JS "faux-async" in scare quotes because in-browser JS (in my experience) isn't actually async, it still runs everything in a single thread, but it has many of the same properties of async: namely, that things like Ajax tend to happen after you expect. I'm not actually sure that if you started multiple ajax requests to the exact same endpoint (or any "blocking" request that took a variable amount of time), that they would return in an unpredictable order like async would imply. I think they would always return the first, then the second. Even if the first one had a longer sleep. 

Damnit. Now I want to try it to find out... Curse you.